frontend/kendra-button-front/src/components/Authenticator.tsx

```diff
@@ -1,57 +1,120 @@
+  const [screen, setScreen] = useState(AuthState.SignIn);
+
+  const checkUser = async (retry, tryCnt = 1): Promise<void> => {
+    const tryLimit = 3;
+    console.log('[checkUser retryflag]', retry);
+    try {
+      const user = await Auth.currentAuthenticatedUser();
+      console.log('[user:Auth]', user);
+      setScreen(AuthState.SignedIn);
+      setUser(user);
+    } catch (e) {
+      console.log('[error in checkUser]', e);
+      if (retry === true && tryLimit > tryCnt) {
+        setTimeout(() => {
+          checkUser(retry, tryCnt + 1);
+        }, 1000);
+      }
+    }
+  };
 
   useEffect(() => {
-    return onAuthUIStateChange((newAuthState, user) => {
-      onStateChange(newAuthState, user);
+    checkUser(true);
+    // intermittently failure
+    // issue that describes same symptoms: https://github.com/aws-amplify/amplify-js/issues/6155#issue-644662860
+    // only error occurs in development
+    Hub.listen('auth', (data) => {
+      console.log('[Hub] data', data);
+      switch (data.payload.event) {
+        case 'signIn':
+          setScreen(AuthState.SignedIn);
+          // setUser(data.payload.event);
+          checkUser(false);
+          break;
+        case 'signIn_failure':
+          console.log('[Hub] signIn_failure');
+          break;
+        default:
+          break;
+      }
     });
   }, []);
 
-  const bgClass = isLoggedIn ? `` : `bg-dark`;
+  const toSignUp = (): void => {
+    setScreen(AuthState.SignUp);
+  };
+  const toSignInGoogle = async (): Promise<void> => {
+    try {
+      //@ts-ignore
+      const res = await Auth.federatedSignIn({ provider: 'Google' });
+      console.log('res', res);
+      setUser(res);
+    } catch (e) {
+      console.log('[error in google]', e);
+    }
+  };
+  const toSignInFacebook = async (): Promise<void> => {
+    try {
+      //@ts-ignore
+      const res = await Auth.federatedSignIn({ provider: 'Facebook' });
+      console.log('res', res);
+      setUser(res);
+    } catch (e) {
+      console.log('[error in facebook]', e);
+    }
+  };
+
+  const bgClass = screen === AuthState.SignIn ? `` : `bg-dark`;
   return (
     <div
       className={`fullscreen ${bgClass} d-flex justify-content-center align-items-center`}
     >
-      <AmplifyAuthenticator usernameAlias="email">
-        <AmplifySignUp
-          slot="sign-up"
-          usernameAlias="email"
-          formFields={[
-            {
-              type: 'email',
-              label: 'Email',
-              placeholder: 'input email',
-              required: true,
-            },
-            {
-              type: 'password',
-              label: 'Password',
-              placeholder: 'input password',
-              required: true,
-            },
-          ]}
-        />
-        {children}
-      </AmplifyAuthenticator>
+      {screen === AuthState.SignedIn ? (
+        children
+      ) : screen === AuthState.SignUp ? (
+        <div> <SignUp setScreen={setScreen} /> </div>
```

Comment:
<div>가 현재로써는 불필요해 보여서 Fragment로만 감싸도 될 것 같습니다

```suggestion
        <> <SignUp setScreen={setScreen} /> </>
```